ospfd/ospf_vty.c

```diff
@@ -9888,12 +9897,10 @@ static int config_write_interface_one(struct vty *vty, struct ospf *ospf)
 static int config_write_interface(struct vty *vty)
 {
 	int write = 0;
-	struct ospf *ospf = NULL;
-	struct listnode *node = NULL;
+	struct ospf *ospf = ospf_lookup_by_vrf_id(VRF_DEFAULT);
 
-	/* Traverse all ospf [vrf] instances */
-	for (ALL_LIST_ELEMENTS_RO(om->ospf, node, ospf))
-		write += config_write_interface_one(vty, ospf);
+	/* Pass Default ospf instances to display MI-OSPF instance id */
+	write += config_write_interface_one(vty, ospf);
```

Comment:
You could loop throughout all VRFs here instead of doing it in the `config_write_interface_one()` function, which already has too many indentation levels.

Also, doesn't `ospf_lookup_by_vrf_id()` need to be called for each VRF? Or maybe you could get the OSPF instance from `vrf->info` inside the VRF loop.